File: ERPServer/Shared/Shared/Entities/Sales/SalesBillHeaderDTO.cs
using Data.Entities.Shared;
using Newtonsoft.Json;
using Shared.Entities.Shared;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Shared.Entities.Sales
{
    public class SalesBillHeaderDTO : AuditEntityDTO
    {
        public bool IsActive { get; set; }
        public string Number { get; set; }
        public string Date { get; set; }
        public decimal Total { get; set; }
        public decimal Discount { get; set; }
        public decimal Transfer { get; set; }
        public decimal TotalAfterDiscount { get; set; }
        public decimal TotalDiscount { get; set; }
        public decimal Paid { get; set; }
        public decimal Remaining { get; set; }
        public string Notes { get; set; }
        public long ClientVendorId { get; set; }
        public string ClientVendorName { get; set; }
        public long? CompanyId { get; set; }

        public virtual List<SalesBillDetailDTO> SalesBillDetailList { get; set; }

    }
}

File: erp-app/src/app/modules/sales/models/sales-bill-header.dto.ts
import { PaymentMethodEnum } from "src/app/shared/enums/payment-method.enum";
import { SalesBillDetailsDTO } from "./sales-bill-details.dto";
import { AuditDTO } from "src/app/shared/models/audit-dto.model";

export class SalesBillHeaderDTO extends AuditDTO {
    number: string;
    isActive: boolean;
    date: string ;
    clientVendorId: number | null;
    clientVendorName: string;
    notes: string;
    companyId: number;
    createdByProfileName: string;
    modifiedByProfileName: string;
    representiveId: number;
    discount: number = 0 | 0;
    otherExpenses: number = 0 | 0;
    total: number = 0 | 0;
    totalAfterDiscount: number = 0 | 0;
    vatAmount: number = 0 | 0;
    totalAfterVAT: number = 0 | 0;
    taxPercentage: number = 0 | 0;
    taxAmount: number = 0 | 0;
    totalAmount: number = 0 | 0;
    paid: number|null ;
    remaining: number = 0 | 0;
    profit: number = 0 | 0;
    isTax: boolean = false;
    isTemp: boolean = false;
    isReturned: boolean = false;
    isNewReturned: boolean = false;
    paymentMethodId: PaymentMethodEnum;
    refNo: string;
    salesBillDetailList: Array<SalesBillDetailsDTO> = Array<SalesBillDetailsDTO>();
    removedSalesBillDetailList: Array<SalesBillDetailsDTO> = Array<SalesBillDetailsDTO>();
	changeProductPriceFromSales: boolean;
	isTempPageMode: boolean;


}